plugins/modules/ec2_vpc_endpoint_info.py

```diff
@@ -176,11 +180,26 @@ def main():
     except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
         module.fail_json_aws(e, msg='Failed to connect to AWS')
 
+    query = module.params.get('query')
+    if query == 'endpoints':
+        module.deprecate('The query option has been deprecated and'
+                         ' will be removed after 2022-12-01',
```

Comment:
Nit, but it would be nice to specify here that in the future this will be the default behaviour.